translator/trace/jaegerthrift_to_protospan_test.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tracetranslator
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/census-instrumentation/opencensus-service/internal/testutils"
+
+	"github.com/jaegertracing/jaeger/thrift-gen/jaeger"
+)
+
+func TestJaegerThriftBatchToOCProto(t *testing.T) {
+	const numOfFiles = 2
+	for i := 1; i <= 2; i++ {
+		thriftInFile := fmt.Sprintf("./testdata/thrift_batch_%02d.json", i)
+		jb := loadJaegerThriftBatchFromJSON(t, thriftInFile)
+		octrace, err := JaegerThriftBatchToOCProto(jb)
+		if err != nil {
+			t.Errorf("Failed to handled Jaeger Thrift Batch from %q. Error: %v", thriftInFile, err)
```

Comment:
If this fails, below we'll have a nil dereference and a bunch of panics. Please add a continue after this error.

Reply:
Ops, my naive golang assumption that it stopped the test right there, I will have to sprinkle `continue` in some other ifs (no panic but no point in continuing the test in such cases).